### PR TITLE
CI: Upgrade mypy to 0.990

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -1646,10 +1646,12 @@ def __array_ufunc__(self, ufunc: np.ufunc, method: str, *inputs, **kwargs):
 
 
 class ExtensionArraySupportsAnyAll(ExtensionArray):
-    def any(self, *, skipna: bool = True) -> bool:
+    def any(self, *, skipna: bool = True) -> bool:  # type: ignore[empty-body]
```
Comment (rhshadrach) [inline pandas/core/arrays/base.py:1649]: Another option I think is to make these abstract methods - I haven't checked if that's viable, and maybe not something you'd want to handle here.

Comment (phofl) [reply pandas/core/arrays/base.py:1649]: Would be ok with doing that, but would like to keep things separated

**Comment by jbrockmendel** [comment]
i upgraded mypy to 0.990 the other day and now im getting

```
% mypy pandas
/usr/local/lib/python3.10/site-packages/dask/utils.py:366: error: Parenthesized context managers are only supported in Python 3.9 and greater  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

**Review by mroeschke — APPROVED** [review]
LGTM @rhshadrach merge when ready

**Review by rhshadrach — APPROVED** [review]

**Comment by rhshadrach** [comment]
Thanks @phofl

**Pull request merged**